File: backend/app/models/generated_file.py
from datetime import datetime

from pydantic import BaseModel


class GeneratedFileMetadata(BaseModel):
    path: str
    artifact_id: str
    size_bytes: int
    language: str | None = None
    updated_at: datetime


class GeneratedFileContent(BaseModel):
    path: str
    content: str
    language: str | None = None

File: frontend/src/types/generatedFile.ts
export type GeneratedFileMetadata = {
  path: string;
  artifact_id: string;
  size_bytes: number;
  language: string | null;
  updated_at: string;
};

export type GeneratedFileContent = {
  path: string;
  content: string;
  language: string | null;
};
